backend-api/src/functions/asyncActiveSession/asyncActiveSessionHandler.test.ts

```diff
@@ -263,28 +263,87 @@ describe("Async Active Session", () => {
+            dependencies,
+            event,
+          );
+
+          expect(
+            mockLoggingAdapter.getLogMessages()[1].logMessage.message,
+          ).toBe("INTERNAL_SERVER_ERROR");
+          expect(mockLoggingAdapter.getLogMessages()[1].data).toStrictEqual({
+            errorMessage: "Mock error when getting session details",
+          });
+          expect(result).toStrictEqual({
+            headers: { "Content-Type": "application/json" },
+            statusCode: 500,
+            body: JSON.stringify({
+              error: "server_error",
+              error_description: "Server Error",
+            }),
+          });
         });
+      });
 
-        const result: APIGatewayProxyResult = await lambdaHandlerConstructor(
-          dependencies,
-          event,
-        );
-
-        expect(mockLoggingAdapter.getLogMessages()[0].logMessage.message).toBe(
-          "STARTED",
-        );
-        expect(mockLoggingAdapter.getLogMessages()[1].logMessage.message).toBe(
-          "COMPLETED",
-        );
+      describe("Given an active session is not found", () => {
+        it("Returns 404 Not Found", async () => {
+          const jwtBuilder = new MockJWTBuilder();
+          const event = buildRequest({
+            headers: { Authorization: `Bearer ${jwtBuilder.getEncodedJwt()}` },
+          });
+          dependencies.sessionService = () =>
+            new MockSessionServiceGetNullSuccessResult();
+
+          const result: APIGatewayProxyResult = await lambdaHandlerConstructor(
+            dependencies,
+            event,
+          );
+
+          expect(result).toStrictEqual({
+            headers: { "Content-Type": "application/json" },
+            statusCode: 404,
+            body: JSON.stringify({
+              error: "session_not_found",
+              error_description:
+                "No active session found for the given sub identifier",
+            }),
+          });
+        });
+      });
 
-        expect(result).toStrictEqual({
-          statusCode: 200,
-          body: "Hello, World",
+      describe("Given an active session is found", () => {
+        it("Returns 200 and the session details", async () => {
+          const jwtBuilder = new MockJWTBuilder();
+          const event = buildRequest({
+            headers: { Authorization: `Bearer ${jwtBuilder.getEncodedJwt()}` },
+          });
+          dependencies.sessionService = () =>
+            new MockSessionServiceGetSuccessResult();
+
+          const result: APIGatewayProxyResult = await lambdaHandlerConstructor(
+            dependencies,
+            event,
+          );
+
+          expect(result).toStrictEqual({
+            headers: { "Content-Type": "application/json" },
+            statusCode: 200,
+            body: JSON.stringify({
+              sessionId: "mockSessionId",
+              redirectUri: "https://mockUrl.com/redirect",
+              state: "mockState",
```

Comment:
Question, is the state value returned here the `sessionState` or `state` from the session? If `sessionState` we could have the value here `mockSessionState` to make it a bit more obvious for people where that value you came from?

If it comes from `state` rather than `sessionState` then no issue with this

Reply:
Mock response has been fixed